testomania/src/main/java/com/earth/testomania/home_screen/presentation/HomeScreen.kt

```diff
@@ -236,6 +255,77 @@ fun CardButton(
     }
 }
 
+@Composable
+fun NetworkStateManager(
+    color: Color,
+    networkConnectivityObserver: NetworkConnectivityObserver,
+    initialNetworkState: Boolean
+) {
+    val status by networkConnectivityObserver.observe()
+        .collectAsState(initial = initialNetworkState)
+
+    if (status == ConnectivityObserver.ConnectionState.Unavailable) {
+        Pulsating {
+            Column(
+                Modifier
+                    .fillMaxSize()
+                    .padding(
+                        start = 26.dp,
+                        top = 80.dp,
+                        end = 26.dp,
+                        bottom = 0.dp
+                    )
+            ) {
+                Box(
+                    Modifier
+                        .fillMaxWidth()
+                        .height(44.dp)
+                        .background(
+                            color = color,
+                            shape = RoundedCornerShape(8.dp)
+                        )
+                        .padding(4.dp)
+                ) {
+                    Image(
+                        painter = painterResource(id = R.drawable.ic_no_connection),
+                        contentDescription = ""
+                    )
+                    Text(
+                        modifier = Modifier
+                            .fillMaxSize()
+                            .padding(top = 6.dp),
+                        textAlign = TextAlign.End,
+                        text = stringResource(R.string.check_your_connection),
+                        color = Color.Black,
```

Comment:
You do not need to specify color, cause light/dark mode for this text will be broken. kiwi does it automatically for light/dark mode